server/routes.go

```diff
@@ -730,6 +683,9 @@ func (s *Server) EmbedHandler(c *gin.Context) {
 				embedding = normalize(embedding[:req.Dimensions])
 			}
 			embeddings[i] = embedding
+			mu.Lock()
+			totalTokens += tokenCount
```

Comment:
I believe you can use `atomic.AddUint64(&totalTokens, int64tokenCount))` here https://dlintw.github.io/gobyexample/public/atomic-counters.html to avoid the manual mutex